_posts/2024-08-10-integration-for-kw-ssh.md

```diff
@@ -0,0 +1,112 @@
+---
+layout: post
+title: Integration Testing for kw-ssh
+date: 2024-07-30 10:30:00
+author: Aquila Macedo
+author_website: https://aquilamacedo.github.io/
+tags: [kw, gsoc, integration_testing, kw-ssh]
+---
+
+`kw-ssh` is a feature in kworkflow that simplifies remote access to machines
+via SSH. It allows you to execute commands or bash scripts on a remote machine
+easily. Additionally, this feature supports file and directory transfer between
+the local and remote machines.
+
+# Overview of kw-ssh Testing Architecture 
+
+![Picture](/images/kw-ssh-illustration.png)
+
+This image illustrates the structure of the integration tests for the `kw-ssh`
+feature, using containers to simulate different operating system environments.
+This setup involves one container acting as the test environment, within which
+tests are executed across three Linux distributions: **Debian**, **Fedora**,
+and **Archlinux**.
+
+Within this test environment, there is a second container, represented in the
+image as "nested" which hosts the SSH server needed for the tests. This
+configuration allows for the isolation of the test environment and execution of
+`kw-ssh` commands on a simulated SSH server, without affecting the local system
+or other containers.
+
+By using containers for each Linux environment and for the SSH server, we
+ensure that tests are conducted in controlled environments, avoiding
+contamination between tests and maintaining result consistency. This approach
+allows the functionality of `kw-ssh` to be validated across different
+distributions, ensuring the code performs as expected on various platforms.
+
+# Details of the Testing Environment
+
+Inside the container that serves as the test environment, I copy a file from
+the host machine, which is the `Containerfile` responsible for generating the
+container with the SSH server. This SSH container is essential for enabling
+connection tests using `kw-ssh`, ensuring that the authentication and transfer
+processes work correctly.
+
+![Picture](/images/containerfile_ssh.png)
```

Comment:
If it isn't too much of a hassle, I recommend adding the code snippet in-line instead of as a screenshot. If this is viable, also do for the other snippets.